code/modules/power/singularity/field_generator.dm

```diff
@@ -317,6 +323,8 @@ field_generator power level display
 	for (var/F in fields)
 		qdel(F)
 
+	shield_floor(FALSE)
+
 	for(var/CG in connected_gens)
 		var/obj/machinery/field/generator/FG = CG
 		FG.connected_gens -= src

```

Comment:
Just create the shield effects in setup_field.

Reply:
The floor shield is created/ torn down when the fields are setup/torn down already.